h/static/scripts/controllers/search-bar-controller.js

```diff
@@ -73,11 +78,56 @@ class SearchBarController extends Controller {
     };
 
     var openDropdown = () => {
+      clearActiveDropdownItem();
+
+      // Reset dropdown items to not hidden
+      Object.keys(this._dropdownItems).forEach(function(key) {
+        this._dropdownItems[key].classList.remove('is-hidden');
+      }.bind(this));
+
       this.setState({open: true});
+
       this._input.addEventListener('keydown', setupListenerKeys,
         true /*capture*/);
     };
 
+    var getVisibleDropdownItems = () => {
+      return this._dropdown.querySelectorAll('li:not(.is-hidden)');
+    };
+
+    var maybeOpenOrCloseDropdown = () => {
```

Comment:
I noticed that this function is actually doing two things: 1. Maybe either opening or closing the dropdown and 2. Showing and hiding all the dropdown items. I wonder if it should be split into two functions?